### ISSUE TITLE
Make the checker report a better message for the case where a CSS system font keyword isn’t the sole value in an expression

### ISSUE THREAD

**Issue by Croydon** [validator/validator#661]
"message-box" is a valid value for the font attribute, but still it is getting display as an error

https://www.w3.org/TR/css-fonts-3/#font-prop

> error: CSS: "font": "message-box" is not a "font" value.

**Comment by sideshowbarker** [comment]
Thanks for reporting this. Do you know which browsers support `message-box` so far?

**Comment by Croydon** [comment]
Firefox does (and has for a long time with -moz-message-box). Beyond that I'm not sure as I'm using this right now only within a Firefox add-on.

Why is this important when it is in the standard?

**Comment by sideshowbarker** [comment]
> Why is this important when it is in the standard?

The level of browser support is important to me personally in determining whether it’s worth me investing time at this point in implementing support for it, and useful enough in general to other users relative to the N other features I haven’t been able to make time to implement support for yet.

**Comment by sideshowbarker** [comment]
Can you please provide a test case that causes the checker to emit that `error: CSS: "font": "message-box" is not a "font" value.` message?

I can’t reproduce it.

Inspecting the source for the CSS-checking backend, I find that that the source already has support for `message-box`.

And I find that the checker doesn’t emit any error for the following test case: 

```css
p { font: message-box }
```

See https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%2C%253C%2521DOCTYPE%2520html%253E%250D%250A%253Chtml%2520lang%253D%2522%2522%253E%250D%250A%253Ctitle%253ETest%253C%252Ftitle%253E%250D%250A%253Cstyle%253Ep%2520%257B%2520font%253A%2520message-box%2520%257D%253C%252Fstyle%253E#textarea

**Comment by Croydon** [comment]
So this

> body {  font: 3.5mm message-box,serif,helvetica,sans-serif; }

throws the error.

While the following examples are all green:

> body {  font: message-box; }

> body {  font-family: message-box; }

> body {  font-family: message-box; font-size: 3.5mm; }

> body {  font-family: message-box, tahoma,arial,helvetica,sans-serif; font-size: 3.5mm; }

> body {  font: 3.5mm tahoma,message-box,arial,helvetica,sans-serif; }

> body {  font: 3.5mm tahoma,arial,helvetica,sans-serif,message-box; }

Seems like **all system fonts keywords** are just not allowed as the first `font-family` value in `font` when having a `font-size` at the same time.

______

> body {  font: 3.5mm caption,tahoma,arial,helvetica,sans-serif,message-box; }

> Error: font: caption is not a font value.

> body {  font: caption; }

> Document checking completed. No errors or warnings to show.

**Comment by sideshowbarker** [comment]
> So this
> 
>     body {  font: 3.5mm message-box,serif,helvetica,sans-serif; }
> 
> throws the error

OK yeah you don’t want to do that. It’s not expressing what you think it is.

Take a look at https://drafts.csswg.org/css-fonts-3/#font-prop:

> Value: `[ [ <‘font-style’> || <font-variant-css21> || <‘font-weight’> || <‘font-stretch’> ]? <‘font-size’> [ / <‘line-height’> ]? <‘font-family’> ] | caption | icon | menu | message-box | small-caption | status-bar`

That says, If you want to specify one of the `caption | icon | menu | message-box | small-caption | status-bar` system fonts, then the system-font keyword you want must be the *only* value in the expression. That is:

```css
body { font: message-box; }
```

That system-font keyword can’t be preceded by anything in the value (e.g., not `3.5mm message-box`), and it can’t be followed by anything.

---

When you instead do `body {  font: 3.5mm message-box,serif,helvetica,sans-serif; }`, what it expresses is that you want to use the (nonexistent) font family with the literal name "message-box". But since there is no font family named `message-box`, the expression will cause the CSS engine to ignore that "message-box" font-family and instead use "serif" as the font family.

There’s a sentence at https://drafts.csswg.org/css-fonts-3/#font-prop which explains:

> Note that the keywords used for the system fonts listed above are only treated as keywords when they occur in the initial position, in other positions the same string is treated as part of the font family name:
> ```css
> font: menu;        /* use the font settings for system menus */
> font: large menu;  /* use a font family named "menu" */
>```

---

All that said, the `body {  font: 3.5mm message-box,serif,helvetica,sans-serif; }` is actually valid per the spec. So strictly speaking, it’s not an error. But the code in the CSS checker backend treats it as an error for the reason explained above; that is, if you’re using `message-box` but it’s not the only value in the expression, then it’s almost certainly a mistake (and not instead you intentionally specifying a font-family with the literal name “message-box”).

So the purpose of having the checker report an error for that case is so you can catch it and fix it.

However, we should make the checker it a much better error message for that case. The current `"message-box" is not a "font" value` isn’t very helpful at all for making clear what the problem is.

So I’m make a change for this in the checker backend, to have it emit a better error message.

---

**Comment by sideshowbarker** [comment]
OK I’ve made a fix and pushed it to https://validator.w3.org/nu/. It causes the checker to report this:

> `font`: No other property values should be used with `message-box`. It should be the sole value; e.g., `p { font: message-box; }`

**Comment by Croydon** [comment]
Thank you!

**Comment by sideshowbarker** [comment]
Thanks for taking time to report this so the behavior could be improved
